### PR TITLE
Painless: Complete Removal of Painless Type

### PR DESCRIPTION
**Pull request by jdconrad**
This completes the removal of Painless Type.  The new data structures in the definition are a map of names (String) to Java Classes and a map of Java Classes to Painless Structs.  The names to Java Classes map can contain a 2 to 1 ratio of names to classes depending on whether or not a short (imported) name is used.  The Java Classes to Painless Structs is 1 to 1 always where the Java Class name must match the Painless Struct name. This should lead a significantly simpler type system in Painless moving forward since the Painless Type only held redundant information since Painless does not support generics.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/DefBootstrap.java [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
 import org.elasticsearch.common.SuppressForbidden;
 
+import java.lang.invoke.CallSite;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.MutableCallSite;
+import java.lang.invoke.WrongMethodTypeException;
+
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/DefBootstrap.java]: The license header seems to be messed up in this file. It should be before the package statement.

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/DefBootstrap.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/DefBootstrap.java]: Huh, this file hasn't been modified much so it's probably legacy.  Will fix.

Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/DefBootstrap.java]: Fixed.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!  Will check in as soon as CI passes.

**Pull request merged**